tests/integration/targets/cpanm/tasks/main.yml

```diff
@@ -0,0 +1,47 @@
+# (c) 2020, Berkhan Berkdemir
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: install perl development package for Red Hat family
+  package:
+    name:
+      - perl-devel
+      - perl-App-cpanminus
+    state: present
+  become: yes
+  when: ansible_os_family == "RedHat"
+
+- name: install perl development package for Debian family
+  package:
+    name:
+      - cpanminus
+    state: present
+  become: yes
+  when: ansible_os_family == "Debian"
+
+- name: install a Perl package
+  cpanm:
+    name: JSON
+    notest: yes
+  register: install_perl_package_result
+  ignore_errors: yes
```

Comment:
Is it ok for this task to fail? What is the reason of ignore errors of this task and assert that it's not failed in next task?

Reply:
Technically I could remove that line off this one play, but the reason for doing that here is that we want it to fail(or not) when doing the assert (below).

Reply:
This is mandatory when testing things that *must* fail.

Reply:
```suggestion
```